### PR TITLE
Test read condition

### PR DESCRIPTION
**Pull request by esteve**
This PR adds an acceptance test for repeated requests to a service.

Connects to ros2/rclcpp#52

Depends on #42

### CODE REVIEW

File: test_rclcpp/CMakeLists.txt [hunk 1]
```diff
@@ -14,13 +14,20 @@ find_package(ament_cmake REQUIRED)
 
 if(AMENT_ENABLE_TESTING)
   find_package(ament_lint_auto REQUIRED)
+  find_package(ament_cmake_gtest REQUIRED)
```
Comment (dirk-thomas) [inline test_rclcpp/CMakeLists.txt]: This line is not necessary since the next line does that already.

File: test_rclcpp/CMakeLists.txt [hunk 2]
```diff
@@ -175,6 +182,38 @@ if(AMENT_ENABLE_TESTING)
...
+      "${middleware_impl}"
+      "rclcpp")
+    add_executable(test_services_server_cpp__${middleware_impl} "test/test_services_server.cpp")
+    add_dependencies(test_services_server_cpp__${middleware_impl} ${PROJECT_NAME})
+    rosidl_target_interfaces(test_services_server_cpp__${middleware_impl}
+      ${PROJECT_NAME} ${typesupport_impl})
+    ament_target_dependencies(test_services_server_cpp__${middleware_impl}
+      "${middleware_impl}"
+      "rclcpp")
+    get_target_property(TEST_SERVICES_SERVER_EXECUTABLE test_services_server_cpp__${middleware_impl} LOCATION)
```
Comment (dirk-thomas) [inline test_rclcpp/CMakeLists.txt]: This should do the same as ament/ament_cmake#29

Same below.

File: test_rclcpp/CMakeLists.txt (hunk 1 above)
Comment (esteve) [inline test_rclcpp/CMakeLists.txt]: Fixed in the `test-services` branch.

File: test_rclcpp/CMakeLists.txt (hunk 2 above)
Comment (esteve) [inline test_rclcpp/CMakeLists.txt]: Fixed in the `test-services` branch.

File: test_rclcpp/test/test_services_client.cpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    executor, node, result1);
+
+  EXPECT_EQ(15, result1.get()->sum);
+
+  EXPECT_EQ(19, result2.get()->sum);
+}
+
+int main(int argc, char ** argv)
+{
+  rclcpp::init(argc, argv);
```
Comment (dirk-thomas) [inline test_rclcpp/test/test_services_client.cpp]: Pass `0, nullptr` instead of `arc, argv`.

Same below.

**Comment by dirk-thomas** [comment]
I would propose to close this in favor of ros2/system_tests#57.

**Pull request closed without merging**